lib/src/components/month_view_components.dart

```diff
@@ -0,0 +1,180 @@
+
+import 'common_components.dart';
+
+class CircularCell extends StatelessWidget {
+  /// Date of cell.
+  final DateTime date;
+
+  /// List of Events for current date.
+  final List<CalendarEventData> events;
+
+  /// Defines if [date] is [DateTime.now] or not.
+  final bool shouldHighlight;
+
+  /// Called when user taps on cell.
+  final VoidCallback onTap;
+
+  /// Background color of circle around date title.
+  final Color backgroundColor;
+
+  /// This class will defines how cell will be displayed.
+  /// To get proper view user [CircularCell] with 1 [MonthView.cellAspectRatio].
+  const CircularCell({
+    Key key,
+    this.date,
+    this.events,
+    this.onTap,
+    this.shouldHighlight,
+    this.backgroundColor = Colors.blue,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Center(
+      child: GestureDetector(
+        onTap: onTap,
+        child: CircleAvatar(
+          backgroundColor:
+              shouldHighlight ? backgroundColor : Colors.transparent,
+          child: Text(
+            "${date.day}",
+            style: TextStyle(
+              fontSize: 20,
+              color: shouldHighlight ? Colors.white : Colors.black,
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class FilledCell extends StatelessWidget {
+  /// Date of current cell.
+  final DateTime date;
+
+  /// List of events on for current date.
+  final List<CalendarEventData> events;
+
+  /// Defines if cell should be highlighted or not.
+  /// If true it will display date title in a circle.
+  final bool shouldHighlight;
+
+  /// Defines background color of cell.
+  final Color backgroundColor;
+
+  /// Called when user taps on any event tile.
+  final void Function(CalendarEventData event, DateTime date) onTileTap;
+
+  /// This class will defines how cell will be displayed.
+  /// This widget will display all the events as tile below date title.
+  ///
+  const FilledCell({
+    Key key,
+    this.date,
+    this.events,
+    this.shouldHighlight,
+    this.backgroundColor,
+    this.onTileTap,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      color: backgroundColor,
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.start,
+        crossAxisAlignment: CrossAxisAlignment.center,
+        mainAxisSize: MainAxisSize.max,
+        children: [
+          SizedBox(
+            height: 5.0,
+          ),
+          shouldHighlight
+              ? CircleAvatar(
+                  radius: 15,
+                  child: Text(
+                    "${date.day}",
+                  ),
+                )
+              : Text(
+                  "${date.day}",
+                ),
+          if (events.isNotEmpty)
+            Expanded(
+                child: Container(
+              margin: EdgeInsets.only(top: 5.0),
+              clipBehavior: Clip.antiAlias,
+              decoration: BoxDecoration(),
```

Comment:
Is there any reason for empty `BoxDecoration()`?

Reply:
Getting this error if I do not provide box decoration.

**The following assertion was thrown building FilledCell(dirty):
'package:flutter/src/widgets/container.dart': Failed assertion: line 273 pos 15: 'decoration != null || clipBehavior == Clip.none': is not true.**